lib/neovim-plugin.nix

```diff
@@ -143,21 +143,26 @@
                       (cfg.packageDecorator cfg.package)
                     ];
                   }
+
                   (lib.optionalAttrs (isColorscheme && (colorscheme != null)) {
                     colorscheme = lib.mkDefault colorscheme;
                   })
+
+                  # Apply any additional configuration added to `extraConfig`
                   (lib.optionalAttrs (args ? extraConfig) (
                     lib.nixvim.modules.applyExtraConfig {
                       inherit extraConfig cfg opts;
                     }
                   ))
                 ]
-                ++ (lib.optionals (!hasConfigAttrs) [
-                  (lib.optionalAttrs callSetup (setLuaConfig setupCode))
-                ])
-                ++ (lib.optionals hasConfigAttrs [
+                # Lua configuration code generation
+                ++ (lib.optionals hasLuaConfig [
+
+                  # Add the plugin setup code `require('foo').setup(...)` to the lua configuration
                   (lib.optionalAttrs callSetup { ${namespace}.${name}.luaConfig.content = setupCode; })
-                  (lib.optionalAttrs (configLocation != null) (setLuaConfig cfg.luaConfig.content))
+
+                  # Write the lua configuration `luaConfig.content` to the config file
+                  (setLuaConfig cfg.luaConfig.content)
```

Comment:
Q: why is it ok to remove the null check? (`configLocation != null`)

Reply:
Because it should never be `null`.
`null` is a forbidden value. Or at least, it is fine as long as `hasLuaConfig` is `false`.